src/hud/Hud.ts

```diff
@@ -0,0 +1,132 @@
+/**
+ * Generate an HTML ID for the HUD element,
+ * with a random 6-digit number to prevent
+ * collisions.
+ * 
+ * @param templateName Name of the HUD element template.
+ * @returns ID for the HTML element
+ */
+function genElementId(templateName: string): string {
+	let output = "";
+	let isUnique = false;
+
+	while (!isUnique) {
+		output = "";
+		[...Array(6)].forEach((_, i) => {
+			output += Math.floor(Math.random() * 10);
+		});
+		if (!(!!document.getElementById("HudElement-" + templateName + "-" + output))) {
+			isUnique = true;
+		}
+	}
+	return "HudElement-" + templateName + "-" + output;
+}
+
+/**
+ * Extend off this class
+ * when making HUD elements.
+ * 
+ * @todo Write a better JSDoc for HudElement lol
+ */
+export class HudElement {
+	private element: HTMLDivElement;
+	private xCoord: number;
+	private yCoord: number;
+
+	constructor(templateName: string) {
+		this.element = document.createElement("div") as HTMLDivElement;
+		this.element.id = genElementId(templateName);
+		this.element.style.position = "fixed"; // Maybe change this later, I'm not a CSS wizard and there could be a better way :/
+		this.element.style.display = "block";
+		this.element.innerHTML = document.getElementById(templateName).innerHTML;
+
+		this.xCoord = 0;
+		this.yCoord = 0;
+	}
+
+	/**
+	 * Set the inner HTMl of the HUD element.
+	 * 
+	 * @param innerHtml The innerHTML value to set.
+	 */
+	setInnerHtml(innerHtml: string) {
+		this.element.innerHTML = innerHtml;
```

Comment:
setting innerHTML should be generally avoided, consider a getter for the element so third party code can manipulate the element instead of replacing it entirely

Reply:
Done